### ISSUE TITLE
Support for other "backends"? (qt etc)

### ISSUE THREAD

**Issue by andrekandore** [stackotter/swift-cross-ui#21]
is it outside of the goals of this project to support other backends such as qt or win32 etc?
or is it strictly gtk for now?

**Comment by stackotter** [comment]
My current plan is to initially just support Gtk, but it shouldn't be too support to allow additional backends to be added. What's the main reason that you might want to be able to choose between multiple backends?

I am hoping that eventually I will find the time to create a custom renderer for SwiftCrossUI using the cross platform WebGPU API or something similar (once it matures). In that case it would be useful to be able to easily add the custom renderer as an experimental backend while it is developed). But I think that won't happen for a long time.

**Comment by stackotter** [comment]
Sorry, that had a bunch of typing mistakes, I fixed them now

**Comment by andrekandore** [comment]
thanks for the quick reply!

> but it shouldn't be too hard to allow additional backends to be added

nice!

> What's the main reason that you might want to be able to choose between multiple backends?

basically i would like to be able to support kde users (myself included) with a more native experience, thats all (^^)

**Comment by stackotter** [comment]
> basically i would like to be able to support kde users (myself included) with a more native experience, thats all (^^)

That seems like a good enough reason :) I’ll see what needs to be done to support custom backends when I next work on SwiftCrossUI. I probably won’t add a qt backend, but once support for other backends is added I’d gladly accept PRs adding new ones! I think it won’t be too hard to add new backends once I add an API for it

**Comment by stackotter** [comment]
Hi @andrekandore, I've finally returned to this project after a long break and I've been making some progress. For now I've decided to keep the project limited to just a Gtk backend (to keep things simple) but I don't have anything against supporting multiple backends per se (e.g. if someone PR'd QT support I'd be all for it).

Hopefully you'll still be able to make some cool stuff with the framework even while using the Gtk backend!
